supervisor.py

```diff
@@ -0,0 +1,158 @@
+"""
+Red-DiscordBot built in Supervisor
+Coded by CCubed (AKA Rory in DAPI)
+"""
+import os
+import subprocess
+import sys
+import urllib.request
+from html.parser import HTMLParser
+from zipfile import *
+
+
+IS_X8664 = sys.maxsize > 2**32
+
+
+class LinkParser(HTMLParser):
+    def __init__(self):
+        self.static_download = None
+        super().__init__()
+
+    def handle_starttag(self, tag, attrs):
+        if tag == 'a':
+            for attr in attrs:
+                if attr[0] == 'href':
+                    if 'win32-static.zip' in attr[1]:
+                        self.static_download = attr[1]
+
+
+def check_for_git():
+    """
+    Does git exist? let's find out.
+    :return: True or False
+    """
+    try:
+        subprocess.check_output(['git', '--version'])
+        return True
+    except Exception as e:
+        return False
+
+
+def check_for_ffmpeg():
+    """
+    Does ffmpeg exist? let's find out
+    :return:  True or False
+    """
+    try:
+        subprocess.check_output(['ffmpeg', '-version'])
+        return True
+    except Exception as e:
+        return False
+
+
+class ProcessManager:
```

Comment:
Popen is being used specifically because we need to separate the process groups. If we don't, then when the supervisor is killed, redbot is going to be killed too. This isn't the desired behavior and with run you cannot use the advanced functionality like preexec_fn or creationflags to ensure that the processes have different groups. As for it being class, it wraps holding the python executable path and the current subprocess. We don't need to do that as a class, but five lines to manage a process looks better than moving all of that out of a class and stuffing it into the main block.